servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -335,6 +336,50 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
     }
   }
 
+  @Override
+  public EntriesResponse getNamespaceEntries(
+      String refName,
+      @Nullable String hashOnRef,
+      @Nullable String namespacePrefix,
+      @Nullable Integer depth)
+      throws NessieNotFoundException {
+    WithHash<NamedRef> refWithHash = namedRefWithHashOrThrow(refName, hashOnRef);
+    try {
+      List<EntriesResponse.Entry> entries;
+      try (Stream<EntriesResponse.Entry> s =
+          getStore()
+              .getKeys(refWithHash.getHash())
+              .map(
+                  key ->
+                      EntriesResponse.Entry.builder()
+                          .name(fromKey(key.getValue()))
+                          .type((Type) key.getType())
+                          .build())) {
+        entries =
+            filterEntries(
+                    s,
+                    (namespacePrefix == null || namespacePrefix.isEmpty())
```

Comment:
I am thinking, would it be more efficient to create a utility function that can be used across other classes to check if the value is null first and then check emptiness for every type instead of doing the double check everytime? e.g `isEmpty(Object value)`. If you guys think it make sense, I would happy to add it as my main contribution to Nessie :D

Reply:
I changed the condition a bit so probably not as relevant, but happy  to have some `Preconditions` like utilities in the codebase